plugin/evm/atomic/sync/atomic_sync_summary.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/crypto"
+)
+
+var _ message.Syncable = (*AtomicSyncSummary)(nil)
+
+// Codec is the codec manager that contains the codec for AtomicSyncSummary and
+// other message types that are used in the network protocol. This is to ensure that the codec
+// version is consistent across all message types and includes the codec for AtomicSyncSummary.
+var Codec codec.Manager
+
+func init() {
+	var err error
+	Codec, err = message.NewCodec(AtomicSyncSummary{})
```

Comment:
Why do we register `AtomicSyncSummary` here? We never seem to marshal/unmarshal these as interfaces.

Reply:
You mean why do we take a interface type in `NewCodec` or why do we register `AtomicSyncSummary{}` rather than `&AtomicSyncSummary{}`?

Reply:
why do we take a interface type in NewCodec?

Reply:
https://github.com/ava-labs/coreth/pull/963#discussion_r2112789420